astropy/coordinates/tests/test_representation.py

```diff
@@ -87,6 +88,22 @@ def representation_equal_up_to_angular_type(rep1, rep2):
     return result & components_allclose(rep1, rep2)
 
 
+class TestRadialRepresentation:
+
+    def test_transform(self):
+        rep = RadialRepresentation(distance=10 * u.kpc)
+
+        # a rotation matrix does not work
+        matrix = rotation_matrix(10 * u.deg)
+        with pytest.raises(NotImplementedError, match="scaled identity matrix"):
+            rep.transform(matrix)
+
+        # only a scaled identity matrix
+        matrix = 3 * np.identity(3)
+        newrep = rep.transform(matrix)
+        assert newrep.distance == 30 * u.kpc
+
```

Comment:
Would it be possible to add a check with differentials?

Reply:
Uh oh. It fails when differentials are attached because of the multiplication bug. I guess this should wait until #11470 is in. It can also be re-simplified to ``self * scale``.

Reply:
In!